src/test/java/org/hisp/dhis/antlr/ValidExpressionTest.java

```diff
@@ -0,0 +1,38 @@
+package org.hisp.dhis.antlr;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith( JUnit4.class )
+public class ValidExpressionTest
+{
+    private TestExpressionVisitor visitor = new TestExpressionVisitor();
+
+    @Test
+    public void testValidExpression() {
+        assertEquals( "a", evaluate( "'a'" ) );
+        assertEquals( true, evaluate( "true" ) );
+        assertEquals( "true", evaluate( "'true'" ) );
+        assertEquals( 34.0, evaluate( "34" ) );
+        assertEquals( "34", evaluate( "'34'" ) );
+        assertEquals( "34.0", evaluate( "'34.0'" ) );
+    }
+
+    @Test(expected = ParserException.class)
+    public void testInvalidSyntaxExpression() {
+        evaluate( "'2' >)_ 1" );
+    }
+
+    @Test(expected = ParserExceptionWithoutContext.class)
+    public void testExpressionWithValidSyntaxAndNotSupportedVariable() {
+        evaluate( "2 > #{not_supported}" );
```

Comment:
can we also test expressions with attributes `"2 > A{not_supported_attribute}", "2 > V{not_supported_variable}`".

Reply:
Added